### PR TITLE
Degree of separation

### PR DESCRIPTION
**Pull request by Yasir-Sh**
## Describe your changes
- Adding Breadth first search algorithm to find degree of separation between followed accounts
- Updating the 'show product page' to include the degree of separation in the review if necessary 
## Issue ticket number and link
- #97 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: src/main/resources/templates/showProduct.html [hunk 1]
```diff
@@ -84,10 +84,18 @@ <h2 class="accordion-header">
...
                         </button>
+
                     </h2>
                     <div th:id="${review.reviewID}" class="accordion-collapse collapse ghostwhite-bg-col" data-bs-parent="#reviewAccordion">
                         <div class="accordion-body text-break">
-                            <a th:text="${review?.account?.username}" th:href="'/account/' + ${review?.account?.username}"></a>
+                            <a th:text="${review?.account?.username}" th:href="'/account/' + ${review?.account?.username}" class="h6"></a>
+                            <small>
+                                <div th:switch="${review.account.getSeparation(currentAccount)}" class="text-muted mb-2">
+                                    <div th:case="1">1 degree</div>
```
Comment (juanitarodeloo) [inline src/main/resources/templates/showProduct.html:94]: As a user I wouldn't know what just "1 degree" means so maybe instead maybe we can follow how LinkedIn does it (i.e 1st, 2nd, 3rd). Also it should be shown in a user's account page as well.

Comment (Mayukh-G) [reply src/main/resources/templates/showProduct.html:94]: I'd argue both don't make sense if you don't know what the number means. At least degree says the word.

**Review by Mayukh-G — CHANGES_REQUESTED** [review]
Looks good. I like the algorithm that you use. Make sure the degree is shown on others' account pages as well.

File: src/main/resources/templates/showProduct.html (hunk 1 above)
Comment (Yasir-Sh) [reply src/main/resources/templates/showProduct.html]: Degree now shows on the user's page (user1 visiting user5). Same as before, if degree isn't relevant it wont be shown on the page. 
image

- As far as 1 degree vs 1st, neither convey the idea clearly. The only way to clearly convey the idea would be '1st degree connection' which would look like images below. 
- Alternatively, I can do something in the middle like '1st degree' (most recent push). Let me know if you have a preference.   
 
image
image

**Pull request merged**